infra/docker/server.Dockerfile

```
@@ -4,32 +4,32 @@ RUN apk add --no-cache openssl && \
-    cd server && npx prisma generate && \
-    cd .. && pnpm --filter server build
+RUN cd server && npx prisma generate && \
+    cd .. && pnpm --filter @shared/types build && \
+    pnpm --filter server build
 
-# Production stage
+# --- Production stage ---
 FROM node:20-alpine AS runner
-RUN apk add --no-cache openssl && \
-    corepack enable && corepack prepare pnpm@9 --activate
+RUN apk add --no-cache openssl
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)